ddl/table.go

```diff
@@ -77,14 +77,15 @@ func onCreateView(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error)
 	schemaID := job.SchemaID
 	tbInfo := &model.TableInfo{}
 	var orReplace bool
-	if err := job.DecodeArgs(tbInfo, &orReplace); err != nil {
+	var oldTbInfoID int64
+	if err := job.DecodeArgs(tbInfo, &orReplace, &oldTbInfoID); err != nil {
 		// Invalid arguments, cancel this job.
 		job.State = model.JobStateCancelled
 		return ver, errors.Trace(err)
 	}
 	tbInfo.State = model.StateNone
 	err := checkTableNotExists(t, job, schemaID, tbInfo.Name.L)
-	if err != nil {
+	if err != nil && !orReplace {
```

Comment:
Watch out here. 
1. `checkTableNotExists` will cancel the job if `err != nil`. 
2. And how about err is `ErrDatabaseNotExists`?

Reply:
@crazycs520 I think we can move `job.State = model.JobStateCancelled` outside of func `checkTableNotExists(t, job, schemaID, tbInfo.Name.L)`, PTAL again

Reply:
@AndrewDi , What if `err = ErrDatabaseNotExists`, and `orReplace = true`? Then line #103 or #108 will return error and then retry, then the job will get into Infinite loop. You should check whether the database exists first.

Reply:
if err = ErrDatabaseNotExists, we will never run into https://github.com/pingcap/tidb/blob/f2343d380009fd35a237947733b2d101e32e9f68/ddl/table.go#L103,because `oldTbInfoID` will always be `0`

Reply:
@AndrewDi , If `drop database` in seesion-1, then before drop database finish, `create or replace view` in session-2, then `err = ErrDatabaseNotExists`, and `oldTbInfoID` will not be 0, because when assign value to `oldTbInfoID`, the database have not been droped.

Reply:
@AndrewDi You can simply mock this scene by add a sleep in `onDropSchema` function. then use 2 session.

Reply:
@crazycs520 So It seems like I can add `job.State = model.JobStateCancelled` to get out of this loop when `DropTable` fail with any reason.

Reply:
@AndrewDi , how about this: 
```sql
   if err != nil {
		if infoschema.ErrDatabaseNotExists.Equal(err) || !orReplace {
			job.State = model.JobStateCancelled
			return ver, errors.Trace(err)
		}
	}
```
`DropTableOrView` or `CreateTableOrView` may be fail by another reason such as network, so we can retry when `DropTableOrView` or `CreateTableOrView`.

Reply:
@crazycs520 I do consider this situation, and found this situation may not happened, because the first loop `t.DropTableOrView` will return err and second loop `checkTableNotExists` will cancel the job.

Reply:
@AndrewDi , I use this PR and only add a sleep in `onDropSchema` function, I Recorded video to mock, https://asciinema.org/a/9fdiVXDBfim3i18Ig9E2ms1Rz

Reply:
@crazycs520 You are right, PTAL again, Thanks

Reply:
@AndrewDi , Still have the same problem. you check after DropTableOrView, but doesn't check CreateTableOrView.  
BTW, I prefer to check here(Line#89):
```sql
   if err != nil {
		if infoschema.ErrDatabaseNotExists.Equal(err) || !orReplace {
			job.State = model.JobStateCancelled
			return ver, errors.Trace(err)
		}
	}
```

Reply:
OK, seems reasonable...otherwise I would need to cancel job on both `DropView` and `CreateView`.